apps/admin/backend/src/app.ts

```diff
@@ -672,12 +713,85 @@ function buildApi({
       });
     },
 
-    getWriteInAdjudicationContext(input: {
-      writeInId: string;
-    }): WriteInAdjudicationContext {
-      return getWriteInAdjudicationContext({
+    getCastVoteRecordVoteInfo(input: {
+      cvrId: string;
+    }): CastVoteRecordVoteInfo {
+      return store.getCastVoteRecordVoteInfo({
+        electionId: loadCurrentElectionIdOrThrow(workspace),
+        ...input,
+      });
+    },
+
+    addWriteInRecord(input: {
+      contestId: ContestId;
+      optionId: Id;
+      cvrId: Id;
+      name: string;
+      side?: Side;
+      isUnmarked: boolean;
+    }): Id {
+      return store.addWriteIn({
+        electionId: loadCurrentElectionIdOrThrow(workspace),
+        castVoteRecordId: input.cvrId,
+        contestId: input.contestId,
+        optionId: input.optionId,
+        side: input.side || 'front', // BMD ballots are always on the front.
+        isUnmarked: input.isUnmarked,
+        machineMarkedText: undefined, // VERIFY if this is correct
```

Comment:
I could use advice on if this is the correct value for `machineMarkedText`

Reply:
We do have a flow for BMD ballots through the adjudication flow that @carolinemodic built last year, and we'll have to confirm that it still works after these changes.

Reply:
@adghayes is it possible for there to be a valid UWI with a BMD? This backend logic should only be relevant for UWIs, though we would need to gate it off on the frontend.

Will check on this in our sync, but it seems like UWI wouldn't make sense for BMDs